### PR TITLE
docker-machine-driver-vultr: recreate from scratch

### PR DESCRIPTION
**Pull request by JounQin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by JounQin** [comment]
image

The test bot seems not using the correct formula source at all.

**Comment by JounQin** [comment]
cc @SMillerDev @carlocab 

related #106850

**Comment by Bo98** [comment]
It'll be because it doesn't have the executable bit (`chmod +x`).

However formulae in homebrew-core needs to be built from source - a formula that just installs from a prebuilt binary like this is doing won't be accepted.

Using the formula source (https://github.com/vultr/docker-machine-driver-vultr/archive/refs/tags/v2.1.0.tar.gz) with an install code of `system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"docker-machine-driver-vultr")` will probably get you most of the way there.

**Comment by JounQin** [comment]
@Bo98 Thanks, it should work now.

File: Formula/d/docker-machine-driver-vultr.rb [hunk 1]
```diff
@@ -22,26 +22,16 @@ class DockerMachineDriverVultr < Formula
...
-    ENV["GO111MODULE"] = "auto"
-    (buildpath/"src/github.com/janeczku/docker-machine-vultr").install buildpath.children
-
-    cd "src/github.com/janeczku/docker-machine-vultr" do
-      system "make"
-      bin.install "build/docker-machine-driver-vultr-v#{version}" => "docker-machine-driver-vultr"
-      prefix.install_metafiles
-    end
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "-o", "docker-machine-driver-vultr", "machine/main.go"
+    bin.install "docker-machine-driver-vultr"
```
Comment (JounQin) [inline Formula/d/docker-machine-driver-vultr.rb:30]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

File: Formula/d/docker-machine-driver-vultr.rb [hunk 2]
```diff
@@ -22,26 +22,15 @@ class DockerMachineDriverVultr < Formula
...
-    ENV["GOPATH"] = buildpath
-    ENV["GO111MODULE"] = "auto"
-    (buildpath/"src/github.com/janeczku/docker-machine-vultr").install buildpath.children
-
-    cd "src/github.com/janeczku/docker-machine-vultr" do
-      system "make"
-      bin.install "build/docker-machine-driver-vultr-v#{version}" => "docker-machine-driver-vultr"
-      prefix.install_metafiles
-    end
+    system "go", "build", *std_go_args(ldflags: "-s -w")
```
Comment (SMillerDev) [inline Formula/d/docker-machine-driver-vultr.rb:29]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w"), "./machine"
```

**Comment by JounQin** [comment]
ping @SMillerDev

**Review by SMillerDev — APPROVED** [review]
Thanks @JounQin  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request merged**